src/main/kotlin/lotto/domain/Rank.kt

```diff
@@ -0,0 +1,22 @@
+package lotto.domain
+
+import java.math.BigDecimal
+
+enum class Rank(val matchCount: Int, val prize: BigDecimal) {
+    THREE(3, BigDecimal(5_000)),
+    FOUR(4, BigDecimal(50_000)),
+    FIVE(5, BigDecimal(1_500_000)),
+    SIX(6, BigDecimal(2_000_000_000)),
+    MISS(-1, BigDecimal(0)),
+    ;
+
+    fun totalPrize(count: Int): BigDecimal {
+        return prize.multiply(BigDecimal(count))
+    }
+
+    companion object {
+        fun from(matchCount: Int): Rank? {
```

Comment:
```suggestion
        fun from(matchCount: Int): Rank {
```

저는 찾을 수 없는 Rank가 있다면 예외를 던지고, Nullable한 객체가 아닌, Null이 아닌 객체를 반환할 듯해요.

Null에 대해 안정하다는 게 바로 코틀린의 장점이니까요~